src/IECoreMaya/SceneShapeSubSceneOverride.cpp

```diff
@@ -0,0 +1,1615 @@
+{
+	switch( style )
+		{
+		case RenderStyle::BoundingBox :
+		case RenderStyle::Wireframe :
+		{
+			if( componentMode )
+			{
+				SelectionStateShaders shaders = getComponentWireShaders();
+				return isSelected ? shaders.selected : shaders.unselected;
+			}
+			else
+			{
+				SelectionStateShaders shaders = getWireShaders();
+				return isSelected ? shaders.selected : shaders.unselected;
+			}
+		}
+		case RenderStyle::Solid :
+		case RenderStyle::Textured :
+			{
+				TextureStateShaders shaders = getAssignedSurfaceShaders( object );
+				return style == RenderStyle::Solid ? shaders.untextured : shaders.textured;
+			}
+		default :
+			return nullptr;
+		}
+}
+
+bool componentsSelectable( const MDagPath &path )
+{
+	MHWRender::DisplayStatus displayStatus = MHWRender::MGeometryUtilities::displayStatus( path );
+	bool selectable = displayStatus == MHWRender::kHilite;
+	return selectable;
+}
+
+// Convenience function that should eventually help with also rendering all instances generated by instancers.
+// Currently not in use, but will come in handy when we get to look into MASH support.
+MObjectArray getConnectedInstancers( MObject object )
+{
+	MObjectArray result;
+
+	MStatus status;
+	MObject current;
+
+	MDagPathArray paths;
+	MMatrixArray matrices;
+	MIntArray pathStartIndices;
+	MIntArray pathIndices;
+
+	MItDependencyNodes it( MFn::kInstancer );
+	for( ; !it.isDone(); it.next() )
+	{
+		MObject currentInstancer = it.item();
+		MFnInstancer instFn( currentInstancer );
+
+		MStatus status;
+		MPlug inputsPlug = instFn.findPlug( "inputHierarchy", &status );
+		if( !status )
+		{
+			continue;
+		}
+
+		// We will need to figure out if our SceneShape feeds into this instancer.
+		for( unsigned int i = 0; i < inputsPlug.numElements(); ++i )
+		{
+			MPlug element = inputsPlug.elementByLogicalIndex( i );
+			MPlugArray srcPlugs;
+			element.connectedTo( srcPlugs, true, false );
+
+			bool found = false;
+			for(unsigned int i=0; i < srcPlugs.length(); ++i)
+			{
+				MFnDagNode connectedNode(srcPlugs[i].node());
+				if( connectedNode.object() == object || connectedNode.hasChild( object ) )
+				{
+					result.append( currentInstancer );
+					found = true;
+					break;
+				}
+			}
+
+			if( found )
+			{
+				break;
+			}
+		}
+	}
+
+	return result;
+}
+
+Imath::M44d worldTransform( const IECoreScene::SceneInterface *scene, double time )
+{
+	IECoreScene::SceneInterface::Path p;
+	scene->path( p );
+
+	IECoreScene::ConstSceneInterfacePtr tmpScene = scene->scene( IECoreScene::SceneInterface::rootPath );
+	Imath::M44d result;
+
+	for ( IECoreScene::SceneInterface::Path::const_iterator it = p.begin(); tmpScene && it != p.end(); ++it )
+	{
+		tmpScene = tmpScene->child( *it, IECoreScene::SceneInterface::NullIfMissing );
+		if ( !tmpScene )
+		{
+			break;
+		}
+
+		result = tmpScene->readTransformAsMatrix( time ) * result;
+	}
+
+	return result;
+}
+
+bool sceneIsAnimated( ConstSceneInterfacePtr sceneInterface )
```

Comment:
Pass raw pointers (`const SceneInterface *`) when a function does not need to retain ownership. This has less overhead and also expresses the function's semantics better.

Reply:
Addressed in 1ec828b